syntax = "proto3";

package glt;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}

  rpc SayHelloAgain (HelloRequest) returns (HelloReply) {}

  rpc FetchData (Intmsg) returns (Bytesmsg) {}
}

message Bytesmsg{
  repeated bytes data = 1;
}
message Intmsg{
  int32 data = 1;
}
// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}
